### PR TITLE
Back to url schemes

### PR DESCRIPTION
**Pull request by biafra23**
Handles #

Changes proposed in this pull request:
- According to Apples documentation url schemes are safe to use with [ASWebAuthenticationSession](https://developer.apple.com/documentation/authenticationservices/aswebauthenticationsession)

``` 
ASWebAuthenticationSession ensures that only the calling app’s session receives 
the authentication callback, even when more than one app registers the same callback
URL scheme.
```

### CODE REVIEW

**Review by DmitryBespalov — APPROVED** [review]

File: Jenkinsfile [hunk 1]
```diff
@@ -26,10 +26,9 @@ pipeline {
         CODECOV_TOKEN = credentials('CODECOV_TOKEN')
         WALLETCONNECT_PROJECT_ID_STAGING = credentials('WALLETCONNECT_PROJECT_ID_STAGING')
         WALLETCONNECT_PROJECT_ID_PROD = credentials('WALLETCONNECT_PROJECT_ID_PROD')
-	WEB3AUTH_GOOGLE_CLIENT_ID_STAGING = credentials('WEB3AUTH_GOOGLE_CLIENT_ID_STAGING')
-	WEB3AUTH_GOOGLE_CLIENT_ID_PROD = credentials('WEB3AUTH_GOOGLE_CLIENT_ID_PROD')
-	WEB3AUTH_GOOGLE_VERIFIER_STAGING = credentials('WEB3AUTH_GOOGLE_VERIFIER_STAGING')
-	WEB3AUTH_GOOGLE_VERIFIER_PROD = credentials('WEB3AUTH_GOOGLE_VERIFIER_PROD')
+	WEB3AUTH_GOOGLE_CLIENT_ID = credentials('WEB3AUTH_GOOGLE_CLIENT_ID')
```
Comment (MouazAlzahabi) [inline Jenkinsfile]: I believe we still need to differentiate between Staging/Prod for verifier and Client ID

Comment (MouazAlzahabi) [reply Jenkinsfile]: Even The Schema will be different in this case, no?

File: Multisig/UI/Safe Management/Create Safe/CreateSafeFlow.swift [hunk 2]
```diff
@@ -108,36 +108,36 @@ class CreateSafeFlow: UIFlow, ASAuthorizationControllerPresentationContextProvid
...
-            view.onViewDidLoad = {
-                self.storeKeyAndCreateSafe(key: key, email: email, keyType: .web3AuthGoogle)
-            }
             self.show(view)
-        }
 
-        loginModel.loginWithCustomAuth(caller: navigationController)
-    }
+        } , keyGenerationComplete: { key, email in
+            self.storeKeyAndCreateSafe(key: key, email: email, keyType: .web3AuthGoogle)
```
Comment (MouazAlzahabi) [inline Multisig/UI/Safe Management/Create Safe/CreateSafeFlow.swift]: We need a weak reference for self in this closure

File: Multisig/UI/Safe Management/Create Safe/CreateSafeFlow.swift [hunk 3]
```diff
@@ -108,36 +108,36 @@ class CreateSafeFlow: UIFlow, ASAuthorizationControllerPresentationContextProvid
...
+        
         guard let privateKey = privateKey else {
             App.shared.snackbar.show(message: "Couldn't create private key from: [\(key)]")
             return
         }
         var keyInfo: KeyInfo? = try? KeyInfo.firstKey(address: privateKey.address)
         if keyInfo == nil {
             do {
-
+
```
Comment (MouazAlzahabi) [inline Multisig/UI/Safe Management/Create Safe/CreateSafeFlow.swift]: Extra line ?

File: Multisig/UI/Safe Management/Create Safe/GoogleWeb3AuthLoginModel.swift [hunk 4]
```diff
@@ -3,42 +3,46 @@ import CustomAuth
 import UIKit
 
 class GoogleWeb3AuthLoginModel {
-    var onClose: ((_ key: String?, _ email: String?) -> Void)
+    var authorizationComplete: () -> Void
+    var keyGenerationComplete: ((_ key: String, _ email: String?) -> Void)
```
Comment (MouazAlzahabi) [inline Multisig/UI/Safe Management/Create Safe/GoogleWeb3AuthLoginModel.swift:8]: no need to define parameter names ((String, String?) -> Void) should be enough

File: Multisig/UI/Safe Management/Create Safe/GoogleWeb3AuthLoginModel.swift [hunk 5]
```diff
@@ -3,42 +3,46 @@ import CustomAuth
...
             return true
         }
         return false
     }
 
-    func loginWithCustomAuth(caller: UIViewController) {
+    func loginWithCustomAuth() {
         Task {
-            let sub = SubVerifierDetails(loginType: .web,
+            let sub = SubVerifierDetails(loginType: .installed,
```
Comment (MouazAlzahabi) [inline Multisig/UI/Safe Management/Create Safe/GoogleWeb3AuthLoginModel.swift:25]: Why installed? shouldn't it be safer if we used web?

File: Multisig/UI/Safe Management/Create Safe/GoogleWeb3AuthLoginModel.swift [hunk 6]
```diff
@@ -3,42 +3,46 @@ import CustomAuth
...
                                          loginProvider: .google,
                                          clientId: App.configuration.web3auth.googleClientId,
                                          verifier: App.configuration.web3auth.googleVerifier,
-                                         redirectURL: App.configuration.web3auth.redirectUrl
+                                         redirectURL: App.configuration.web3auth.redirectScheme + "://"
```
Comment (MouazAlzahabi) [inline Multisig/UI/Safe Management/Create Safe/GoogleWeb3AuthLoginModel.swift]: this is the second place were we append "://", I would prefer if we predefine it

File: Jenkinsfile (hunk 1 above)
Comment (biafra23) [reply Jenkinsfile]: Yes, we need prod & staging variants here.

Schema is depending on the client id provided by google. that is correct.

File: Multisig/UI/Safe Management/Create Safe/GoogleWeb3AuthLoginModel.swift (hunk 4 above)
Comment (biafra23) [reply Multisig/UI/Safe Management/Create Safe/GoogleWeb3AuthLoginModel.swift:8]: I find this more readable. And easier to understand. Is there a disadvantage to it?

File: Multisig/UI/Safe Management/Create Safe/GoogleWeb3AuthLoginModel.swift (hunk 5 above)
Comment (biafra23) [reply Multisig/UI/Safe Management/Create Safe/GoogleWeb3AuthLoginModel.swift:25]: Won't work with .web.  
I guess .web needs to match the web oauth clients and .installed matched iOS type.

File: Jenkinsfile (hunk 1 above)
Comment (biafra23) [reply Jenkinsfile:29]: Why do we need _PROD & _STAGING in this file? Why not handle it like the INFURA_KEY?

Comment (biafra23) [reply Jenkinsfile]: Jenkinsfile is updated

**Review by DmitryBespalov — APPROVED** [review]

**Pull request merged**